pkg/configuration/toolchainconfig/configuration.go

```diff
@@ -0,0 +1,373 @@
+package toolchainconfig
+
+import (
+	"strings"
+	"time"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	ToolchainStatusName = "toolchain-status"
+
+	// NotificationDeliveryServiceMailgun is the notification delivery service to use during production
+	NotificationDeliveryServiceMailgun = "mailgun"
+)
+
+var logger = logf.Log.WithName("configuration")
+
+type ToolchainConfig struct {
+	cfg     *toolchainv1alpha1.ToolchainConfigSpec
+	secrets map[string]map[string]string
+}
+
+func NewToolchainConfig(cfg *toolchainv1alpha1.ToolchainConfigSpec, secrets map[string]map[string]string) ToolchainConfig {
+	return ToolchainConfig{
+		cfg:     cfg,
+		secrets: secrets,
+	}
+}
+
+func (c *ToolchainConfig) Print() {
+	logger.Info("Toolchain configuration variables", "ToolchainConfigSpec", c.cfg)
+}
+
+func (c *ToolchainConfig) Environment() string {
+	return getString(c.cfg.Host.Environment, "prod")
+}
+
+func (c *ToolchainConfig) AutomaticApproval() AutoApprovalConfig {
+	return AutoApprovalConfig{c.cfg.Host.AutomaticApproval}
+}
```

Comment:
just curious - do we really need/want to move all the methods to common? wouldn't it be easier to keep here some high-level abstraction and reuse it in host & registration service? I guess that none of the methods of the `ToolchainConfig` type is used in both host & reg-service. I that's true, then we can split it and create a new types in the repos with related functions. Cache and other stuff can stay here.

Reply:
Yeah, you're right. I thought about this as well while making the registration service changes. The registration service only uses the `RegistrationService` section of the ToolchainConfig so let me do some refactoring to try to abstract out the cache and stuff.